python/pyspark/pandas/internal.py

```diff
@@ -987,34 +986,52 @@ def attach_distributed_sequence_column(
                     [], schema=StructType().add(column_name, data_type=LongType(), nullable=False)
                 )
 
-    def spark_column_for(self, label: Label) -> Column:
+    def spark_column_for(self, label: Label) -> PySparkColumn:
         """Return Spark Column for the given column label."""
         column_labels_to_scol = dict(zip(self.column_labels, self.data_spark_columns))
         if label in column_labels_to_scol:
             return column_labels_to_scol[label]
         else:
             raise KeyError(name_like_string(label))
 
-    def spark_column_name_for(self, label_or_scol: Union[Label, Column]) -> str:
+    def spark_column_name_for(self, label_or_scol: Union[Label, PySparkColumn]) -> str:
         """Return the actual Spark column name for the given column label."""
-        if isinstance(label_or_scol, (Column, ConnectColumn)):
+        if is_remote():
+            from pyspark.sql.connect.column import Column as ConnectColumn
+
+            Column = ConnectColumn
+        else:
+            Column = PySparkColumn  # type: ignore[assignment]
+        if isinstance(label_or_scol, Column):
```

Comment:
Can you switch the condition to `Label`? then you can remove if-else

Reply:
Nice!